std/format.d

```diff
@@ -978,6 +978,18 @@ uint formattedRead(R, Char, S...)(ref R r, const(Char)[] fmt, auto ref S args)
     assert(aa3 == ["hello":1, "world":2]);
 }
 
+// test rvalue using
+@system pure unittest
+{
+    string[int] aa1;
+    formattedRead!("%s")(`[1:"hello", 2:"world"]`, &aa1);
```

Comment:
You don't need to take `aa1` by reference here. That's the old style, which we should look into deprecating. The compiler and `formattedRead` are smart enough to automatically do it for you.